common/src/main/java/org/conscrypt/OpenSSLSocketImpl.java

```diff
@@ -830,18 +774,17 @@ public SSLSession getSession() {
                 return SSLNullSession.getNullSession();
             }
         }
-        return Platform.wrapSSLSession(sslSession);
+        return SSLUtils.wrapSSLSession(sslSession);
     }
 
     /* @Override */
-    @SuppressWarnings("MissingOverride")  // For compilation with Java 6.
+    @SuppressWarnings("MissingOverride") // For compilation with Java 6.
     public SSLSession getHandshakeSession() {
-        return handshakeSession;
+        return sslSession;
```

Comment:
I'm kind of nervous about this relaxing a restriction we'll have to live with forever. The Javadoc for this method says:
```
Returns: null if this instance is not currently handshaking, ...
```